psydac/api/ast/linalg.py

```diff
@@ -593,4 +642,86 @@ def transpose_3d( M:'float[:,:,:,:,:,:]', Mt:'float[:,:,:,:,:,:]', n1:"int64", n
     return
 
 _transpose  = {1:transpose_1d,2:transpose_2d, 3:transpose_3d}
-_args_dtype = {1:[repr('float[:,:]')]*2 + [repr('int64')]*11,2:[repr('float[:,:,:,:]')]*2 + [repr('int64')]*22, 3:[repr('float[:,:,:,:,:,:]')]*2 + ['int']*33}
+transpose_args_dtype = {1:[repr('float[:,:]')]*2 + [repr('int64')]*11,2:[repr('float[:,:,:,:]')]*2 + [repr('int64')]*22, 3:[repr('float[:,:,:,:,:,:]')]*2 + [repr('int64')]*33}
+#========================================================================================================
+def interface_transpose_1d( M:'float[:,:]', Mt:'float[:,:]', d_start:"int64", c_start:"int64", dim:"int64" ,n1:"int64", nc1:"int64", gp1:"int64", p1:"int64",
+                            nd1:"int64", ndT1:"int64", si1:"int64", sk1:"int64", sl1:"int64"):
+
+    d1 = gp1-p1
+    for x1 in range(n1):
+        j1 = gp1 + x1
+        for l1 in range(nd1):
+            i1 = si1 + x1 + l1 + d1
+            k1 = sk1 -l1
+            m1 = l1+sl1
+            if k1<ndT1 and k1>-1 and m1<nd1 and i1<nc1:
+                Mt[j1,m1] = M[i1,k1]
+
+# ...
+def interface_transpose_2d( M:'float[:,:,:,:]', Mt:'float[:,:,:,:]', d_start:"int64", c_start:"int64", dim:"int64", n1:"int64", n2:"int64", nc1:"int64", nc2:"int64",
+                           gp1:"int64", gp2:"int64", p1:"int64", p2:"int64",nd1:"int64", nd2:"int64", ndT1:"int64", ndT2:"int64",
+                           si1:"int64", si2:"int64", sk1:"int64", sk2:"int64", sl1:"int64", sl2:"int64"):
+
+    d1 = gp1-p1
+    d2 = gp2-p2
+
+    for x1 in range(n1):
+        for x2 in range(n2):
+            j1 = gp1 + x1
+            j2 = gp2 + x2
+            for l1 in range(nd1):
+                for l2 in range(nd2):
+
+                    i1 = si1 + x1 + l1 + d1
+                    i2 = si2 + x2 + l2 + d2
+
+                    k1 = sk1 -l1
+                    k2 = sk2 -l2
+
+                    m1 = l1+sl1
+                    m2 = l2+sl2
+                    if k1<ndT1 and k1>-1 and k2<ndT2 and k2>-1\
+                        and m1<nd1 and m2<nd2 and i1<nc1 and i2<nc2:
+                        Mt[j1,j2, m1,m2] = M[i1,i2, k1,k2]
+
+# ...
+def interface_transpose_3d( M:'float[:,:,:,:,:,:]', Mt:'float[:,:,:,:,:,:]', d_start:"int64", c_start:"int64", dim:"int64", n1:"int64", n2:"int64", n3:"int64",
+                  nc1:"int64", nc2:"int64", nc3:"int64", gp1:"int64", gp2:"int64", gp3:"int64", p1:"int64", p2:"int64", p3:"int64",
+                  nd1:"int64", nd2:"int64", nd3:"int64",ndT1:"int64", ndT2:"int64", ndT3:"int64", si1:"int64", si2:"int64",
+                  si3:"int64", sk1:"int64", sk2:"int64", sk3:"int64",sl1:"int64", sl2:"int64", sl3:"int64"):
```

Comment:
Given that this code is not automatically generated, can you please split these long lines of arguments at a width of 80-chars?

Reply:
done in [1d09218](https://github.com/pyccel/psydac/pull/160/commits/1d09218d700203883784eb8a5109e576b7f43b60)